File: spark/src/main/java/com.spark/java/sparkCore/WordTest.java
package com.spark.java.sparkCore;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

import java.util.Arrays;

public class WordTest {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setAppName("words");
//                .setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        String path = "data\\words.txt";
        if(args.length == 1){
            path = args[0];
        }
        JavaRDD<String> javaRDD = sc.textFile(path);
        JavaPairRDD<String, Integer> resultRDD = javaRDD.flatMap(s -> Arrays.asList(s.split(" ")).iterator())
                .mapToPair(s -> new Tuple2<>(s, 1))
                .reduceByKey((x, y) -> x + y);

        resultRDD.foreach(e -> System.out.println(e._1 + "~" + e._2));
    }
}

File: spark/src/main/java/com.spark/scala/WordTest.scala
package com.spark.scala

import org.apache.spark.{SparkConf, SparkContext}

object WordTest {

  def main(args: Array[String]): Unit = {

    val conf = new SparkConf()
    conf.setAppName("test")
      .setMaster("local")
    val sc = new SparkContext(conf);

    val array = Array(1,2,3,4,5,6,7,8,9)

    val rdd = sc.parallelize(array)

    val sum = rdd.reduce(_ + _)

    println(sum)
  }

}
